### PR TITLE
Clean up precice-ci.yml

### PR DESCRIPTION
**Pull request by peterrum**
FYI @DavidSCN

### CODE REVIEW

File: .github/workflows/precice-ci.yml [hunk 1]
```diff
@@ -5,31 +5,42 @@ on:
...
+        run: |
+          sudo chown -R $USER:$USER $GITHUB_WORKSPACE
+          
+      - uses: actions/checkout@v2
+
+      - name: Compile
+        run: |
+          sudo apt-get -y update
+          wget https://github.com/precice/precice/releases/download/v2.1.1/libprecice2_2.1.1_focal.deb
+          sudo apt-get -y install ./libprecice2_2.1.1_focal.deb
```
Comment (DavidSCN) [inline .github/workflows/precice-ci.yml:36]:
```suggestion
          wget https://github.com/precice/precice/releases/download/v2.3.0/libprecice2_2.3.0_focal.deb
          sudo apt-get -y install ./libprecice2_2.3.0_focal.deb
```
Considering that you update this, you could also update the related preCICE version if you like.

**Comment by bangerth** [comment]
I don't know anything about how these CI files should be written. This requires someone else's approval.

File: .github/workflows/precice-ci.yml [hunk 2]
```diff
@@ -5,31 +5,42 @@ on:
...
-         
-         echo $command
-         
-         docker pull dealii/dealii:v9.2.0-focal
-         docker run -t dealii/dealii:v9.2.0-focal /bin/sh -c "$command";
+      - name: Setup
+        run: |
+          sudo chown -R $USER:$USER $GITHUB_WORKSPACE
+          
+      - uses: actions/checkout@v2
```
Comment (marcfehling) [inline .github/workflows/precice-ci.yml:30]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/precice-ci.yml [hunk 3]
```diff
@@ -5,31 +5,42 @@ on:
...
+
+      - name: Compile
+        run: |
+          sudo apt-get -y update
+          wget https://github.com/precice/precice/releases/download/v2.3.0/libprecice2_2.3.0_focal.deb
+          sudo apt-get -y install ./libprecice2_2.3.0_focal.deb
```
Comment (marcfehling) [inline .github/workflows/precice-ci.yml:36]:
```suggestion
          wget https://github.com/precice/precice/releases/download/v2.4.0/libprecice2_2.4.0_focal.deb
          sudo apt-get -y install ./libprecice2_2.4.0_focal.deb
```

File: .github/workflows/precice-ci.yml [hunk 4]
```diff
@@ -5,31 +5,42 @@ on:
...
   build:
-
+    name: Build ${{ matrix.build_type }}-dealii:${{ matrix.dealii_version }}
     runs-on: ubuntu-latest
 
+    strategy:
+      fail-fast: false
+      matrix:
+        build_type: ["Debug"]
+        dealii_version: ["master", "v9.3.0"]
```
Comment (marcfehling) [inline .github/workflows/precice-ci.yml:20]:
```suggestion
        dealii_version: ["master", "v9.4.0"]
```

File: .github/workflows/precice-ci.yml (hunk 3 above)
Comment (marcfehling) [inline .github/workflows/precice-ci.yml:36]: These lines would rather fit in the `setup` part above. Do you agree?

**Review by marcfehling — APPROVED** [review]
This looks really nice when you get the docker image in the github-actions script.

**Comment by bangerth** [comment]
@peterrum What's the status here?

**Comment by marcfehling** [comment]
I've solved the merge conflict by rebasing.

The `permission denied` error can be fixed by overriding the default container user, see also https://github.com/actions/checkout/issues/956. I will push a fix in a separate commit.

I will also apply my suggestions in a separate commit. Updating to the latest preCICE 2.5.0 release yields a different output and thus lets the test fail. I will thus update to preCISE 2.4.0 instead.

**Review by peterrum — COMMENTED** [review]
@marcfehling Thanks for the follow-up commits! Looks good to me. I cannot approve my own PR^^

**Comment by marcfehling** [comment]
The PR has been around for a year. Let's merge this.

**Pull request merged**